src/amazon/AmazonClient.js

```diff
@@ -75,17 +115,36 @@ class AmazonClient {
     This is necessary because Amazon's API allows a maximum of 5 items to be looked up at at time.
     For details, refer to idList param at http://docs.developer.amazonservices.com/en_US/products/Products_GetMatchingProductForId.html 
   */
-  _batchedProductsRequest(productIds, idType, delayIndex=0) {
+  _batchedProductsRequest(productIds, idType) {
     let promises = [];
     let index=0;
     let sliceEnd;
-
+    let incrementValue=5;
     do {
-      sliceEnd = sliceEnd > productIds.length ? productIds.length : index + 5;
-      promises.push(this._getProductsById(productIds.slice(index, sliceEnd), idType, index/5));
-      index+=5;
+      sliceEnd = sliceEnd > productIds.length ? productIds.length : index + incrementValue;
+      promises.push(this._getProductsById(productIds.slice(index, sliceEnd), idType, index/incrementValue));
+      index+=incrementValue;
     } while (index < productIds.length);
+    return Promise.all(promises.map(function(promise) {
+      return promise.reflect();
+    }));
+  }
 
+   /*
+    Batches lowest offer requests
+   */
+  _batchedLowestOfferListingsRequest(asinList) {
+    let promises = [];
+    let index=0;
+    let sliceEnd;
+    let incrementValue=10;
+    let itemCondition = 'New';
+    let excludeMe = true;
```

Comment:
The only thing you are doing with some of these variables in this function is passing them to `_getLowestOfferLIstingsByASIN`. If that's the case, then they should be declared in the function that actually uses them, not here. It also seems some of them should be `const`.

Reply:
fixed here  59fd1c1